pkg/monitoring/config.go

```diff
@@ -0,0 +1,207 @@
+// - most params are passed as flags to the binary.
+// - username and passwords can be overriden by environment variables.
+// - feeds configuration can be passed by an RDD url or a local file (useful for testing).
+// ParseConfig also validates and parses some of these inputs and returns an error for the first input that is found incorrect.
+func ParseConfig() (Config, error) {
+
+	cfg := Config{}
+	flag.StringVar(&cfg.Solana.RPCEndpoint, "solana.rpc_endpoint", "", "")
+	flag.StringVar(&cfg.Solana.NetworkName, "solana.network_name", "", "")
+	flag.StringVar(&cfg.Solana.NetworkID, "solana.network_id", "", "")
+	flag.StringVar(&cfg.Solana.ChainID, "solana.chain_id", "", "")
+
+	flag.StringVar(&cfg.Kafka.Topic, "kafka.topic", "", "solana-mainnet")
+	flag.StringVar(&cfg.Kafka.Brokers, "kafka.brokers", "", "")
+	flag.StringVar(&cfg.Kafka.ClientID, "kafka.client_id", "", "")
+	flag.StringVar(&cfg.Kafka.SecurityProtocol, "kafka.security_protocol", "", "")
+	flag.StringVar(&cfg.Kafka.SaslMechanism, "kafka.sasl_mechanism", "", "")
+	flag.StringVar(&cfg.Kafka.SaslUsername, "kafka.sasl_username", "", "")
+	flag.StringVar(&cfg.Kafka.SaslPassword, "kafka.sasl_password", "", "")
+
+	flag.StringVar(&cfg.SchemaRegistry.URL, "schema_registry.url", "", "")
+	flag.StringVar(&cfg.SchemaRegistry.Username, "schema_registry.username", "", "")
+	flag.StringVar(&cfg.SchemaRegistry.Password, "schema_registry.password", "", "")
+
+	var feedsFilePath string
+	var feedsRDDURL string
+	flag.StringVar(&feedsFilePath, "feeds.file_path", "", "")
+	flag.StringVar(&feedsRDDURL, "feeds.rdd_url", "", "")
+
+	flag.StringVar(&cfg.Http.Address, "http.address", "", "")
+
+	flag.Parse()
+
+	for flagName, value := range map[string]string{
+		"-solana.rpc_endpoint": cfg.Solana.RPCEndpoint,
+
+		"-kafka.brokers":           cfg.Kafka.Brokers,
+		"-kafka.client_id":         cfg.Kafka.ClientID,
+		"-kafka.security_protocol": cfg.Kafka.SecurityProtocol,
+
+		"-schema_registry.url": cfg.SchemaRegistry.URL,
+
+		"-http.address": cfg.Http.Address,
+	} {
+		if value == "" {
+			return cfg, fmt.Errorf("flag '%s' is required", flagName)
+		}
+	}
+
+	if value, isPresent := os.LookupEnv("KAFKA_TOPIC"); isPresent {
+		cfg.Kafka.Topic = value
+	}
+	if value, isPresent := os.LookupEnv("KAFKA_SASL_USERNAME"); isPresent {
+		cfg.Kafka.SaslUsername = value
+	}
+	if value, isPresent := os.LookupEnv("KAFKA_SASL_PASSWORD"); isPresent {
+		cfg.Kafka.SaslPassword = value
+	}
+	if value, isPresent := os.LookupEnv("SCHEMA_REGISTRY_USERNAME"); isPresent {
+		cfg.SchemaRegistry.Username = value
+	}
+	if value, isPresent := os.LookupEnv("SCHEMA_REGISTRY_PASSWORD"); isPresent {
+		cfg.SchemaRegistry.Password = value
+	}
+
+	var feeds = []jsonFeedConfig{}
+	if feedsFilePath == "" && feedsRDDURL == "" {
+		return cfg, fmt.Errorf("feeds configuration missing, either '-feeds.file_path' or '-feeds.rdd_url' must be set")
+	} else if feedsRDDURL != "" {
+		res, err := http.Get(feedsRDDURL)
```

Comment:
we shouldn't use default http cause does not have a timeout.

Reply:
Agreed

Reply:
Updated!